public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx

```diff
@@ -281,6 +283,49 @@ const isCortexLokiOrRecordingRule = (watch: UseFormWatch<RuleFormValues>) => {
   return (ruleType === RuleFormType.cloudAlerting || ruleType === RuleFormType.cloudRecording) && dataSourceName !== '';
 };
 
+// the backend will always execute "hidden" queries, so we have no choice but to remove the property in the front-end
+// to avoid confusion. The query editor shows them as "disabled" and that's a different semantic meaning.
+// furthermore the "AlertingQueryRunner" calls `filterQuery` on each data source and those will skip running queries that are "hidden"."
+// It seems like we have no choice but to act like "hidden" queries don't exist in alerting.
+const ignoreHiddenQueries = (ruleDefinition: RuleFormValues): RuleFormValues => {
```

Comment:
maybe it makes more sense to apply this method directly to the ruleDefinition.queries?, I mean having the queries as a param instead of the rule definition?

Reply:
So I decided to apply it to the entire rule definition so I wouldn't have to use mutation, if the input params only took the queries and returned those (no mutation) I would still have to spread them in to the rule definition in three separate places in the code (we allow rule form value population from existing rule, cloned rule, or query params) :)